services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCAuthorizationDetailsProcessor.java

```diff
@@ -221,39 +220,116 @@ private AuthorizationDetailsResponse buildAuthorizationDetailResponse(Authorizat
+        for (String credentialIdentifier : credentialIdentifiers) {
+            // Store the mapping between credential identifier and configuration ID in client session
+            String mappingKey = OID4VCIssuerEndpoint.CREDENTIAL_IDENTIFIER_PREFIX + credentialIdentifier;
+            clientSession.setNote(mappingKey, credentialConfigurationId);
+            logger.debugf("Stored credential identifier mapping: %s -> %s", credentialIdentifier, credentialConfigurationId);
+        }
+
+        // Store claims in user session for later use during credential issuance
         if (detail.getClaims() != null) {
             // Store claims with a unique key based on credential configuration ID
-            String claimsKey = "AUTHORIZATION_DETAILS_CLAIMS_" + credentialConfigurationId;
+            String claimsKey = OID4VCIssuerEndpoint.AUTHORIZATION_DETAILS_CLAIMS_PREFIX + credentialConfigurationId;
             try {
                 userSession.setNote(claimsKey, JsonSerialization.writeValueAsString(detail.getClaims()));
             } catch (Exception e) {
                 logger.warnf(e, "Failed to store claims in user session for credential configuration %s", credentialConfigurationId);
             }
 
-            // Store credential context mapping using credential identifier as key
-            for (String credentialIdentifier : credentialIdentifiers) {
-                String contextKey = "CREDENTIAL_CONTEXT_" + credentialIdentifier;
-                try {
-                    // Store the complete credential context for later retrieval
-                    Map<String, Object> credentialContext = Map.of(
-                            "credentialConfigurationId", credentialConfigurationId,
-                            "claims", detail.getClaims(),
-                            "type", OPENID_CREDENTIAL_TYPE
-                    );
-                    userSession.setNote(contextKey, JsonSerialization.writeValueAsString(credentialContext));
-                } catch (Exception e) {
-                    logger.warnf(e, "Failed to store credential context for identifier %s", credentialIdentifier);
-                }
-            }
-
             // Include claims in response
             responseDetail.setClaims(detail.getClaims());
         }
 
         return responseDetail;
     }
 
+
+    /**
+     * Process authorization_details from the credential offer when authorization_details parameter is not present in the token request.
+     * This method generates authorization_details based on the credential_configuration_ids from the credential offer.
+     *
+     * @param clientSession the client session that contains the credential offer information
+     * @return the authorization details response if processing was successful, null otherwise
+     */
+    public List<AuthorizationDetailsResponse> processFromCredentialOffer(AuthenticatedClientSessionModel clientSession) {
```

Comment:
Does this method generate the authorization-details in case they are missing? If so, then I'd rename the method to like `generateAuthorizationDetailsFromCredentialOffer` to better express it's intent.